python/housinginsights/sources/cama.py

```diff
@@ -0,0 +1,208 @@
+        super().__init__(CamaApiConn.BASEURL)
+
+    def get_data(self):
+        """
+        Grabs data from CAMA. Individual CAMA property retrieves zone_type data
+        from MAR api. Count number of housing units and bedroom units per zone.
+        Return the count data (in dictionary form) to be processed into csv
+        by get_csv() method.
+        """
+
+        mar_api = MarApiConn()
+        result = self.get(urlpath='/c5fb3fbe4c694a59a6eef7bf5f8bc49a_25.geojson', params=None)
+
+        if result.status_code != 200:
+            err = "An error occurred during request: status {0}"
+            raise Exception(err.format(result.status_code))
+        cama_data = result.json()
+
+        """
+        Example of: anc_count = [OrderedDict([('zone_type', 'anc'), ('zone', 'ANC 2B'), 
+                                ('housing_unit_count', 10), ('bedroom_unit_count', 10)], etc)]
+        """
+        zone_types = ['anc', 'census_tract', 'neighborhood_cluster', 'ward', 'zip']
+        anc_count = []
+        census_count = []
+        cluster_count = []
+        ward_count = []
+        zipcode_count = []
+
+
+        """
+        Take each CAMA property data and retrieve the MAR data.
+        """
+        """
+        Certain square values have four digits + a letter. (ex. 8888E)
+        Square would be the first four digits and suffix would be the letter.
+        SSL sometimes comes as 8 digit string without spacing in the middle.
+        """
+        """
+        CAMA data includes bldgs under construction. CAMA's data includes AYB of 2018
+        as of June 2017. We eliminate all data points that are under construction and
+        don't provide any housing units and bedrm at this time.
+        """
+        for row in cama_data['features']:
+            try:
+                current_year = int(datetime.date.today().strftime('%Y'))
+                #Skipping none values for units under construction
+                if row['properties']['AYB'] is not None and int(row['properties']['AYB']) > current_year:
+                    continue
+
+                objectid = row['properties']['OBJECTID']
+                if len(row['properties']['SSL']) == 8:
+                    square = row['properties']['SSL'][:4]
+                    lot = row['properties']['SSL'][4:]
+                else:
+                    square, lot = row['properties']['SSL'].split()
+                suffix = ' '
+                if len(square) > 4:
+                    square = square[:4]
+                    suffix = square[-1]
+
+                mar_return = mar_api.get_data(square, lot, suffix)
+
+                ''' Count the housing units and bedrooms '''
+                num_units = 0
+                if row['properties']['NUM_UNITS']: num_units = row['properties']['NUM_UNITS']
+                if num_units == 0:
+                    num_units = 1
+
+                bedrm = row['properties']['BEDRM']
+                if bedrm == 0: bedrm = 1
```

Comment:
In the CAMA data, there are about 207 data points that have 0 bedrooms. 
- 172 out of 207 have 1 housing unit but contains "0 bedroom". Many has land use code of 011 - residential single family housing. Assumption was made that 1 housing unit should be equivalent to 1 bedroom. 
- There are also data points like above with 0 housing units and 0 bedrooms. I made the assumption that if the MAR data exists then there should be at least 1 housing unit and 1 bedroom.